pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +281,56 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) netContainerCleanup(containerInfo containerGCInfo) {
```

Comment:
@caseydavenport we can't say that at this point.  The net container being cleaned up may well be a dead net container of a still-running pod that has already created a new net container. The sequence of events you post here is exactly one of the sequences I saw when testing this out.

Is there something I could do to mitigate this issue for plugin writers, and/or could it be documented for kube 1.5? Could you describe a bit more how the Calico plugin handles correlation between IPAM and specific container IDs versus pod name/namespace?

Also, does the existing Calico workaround handle the case where the pod does not get rescheduled on the same node, but the net container is still dead with its IP allocation intact?  In that case kubelet would never have called the plugin at all, either for setup or teardown, so the plugin would need some kind of node-external tracking service for the IPAM to know when the pod was rescheduled on another node and then clear out the allocation on the existing node somehow.